src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -808,6 +813,59 @@ def get_disable_fips_image(self) -> bool:
         # read the original value passed by the command
         return self.raw_param.get("disable_fips_image")
 
+    def get_set_localdns(self) -> bool:
+        return self.raw_param.get("set_localdns", False)
+
+    def get_localdns_config(self):
+        return self.raw_param.get("localdns_config")
+
+    def get_localdns_mode(self):
+        return self.raw_param.get("localdns_mode")
+
+    def get_localdns_profile(self):
+        """
+        Returns the local DNS profile dict if set, or None.
+        Handles validation and file loading.
+        """
+        set_localdns = self.get_set_localdns()
+        config = self.get_localdns_config()
+        mode = self.get_localdns_mode()
+        if not set_localdns and (config or mode):
+            raise InvalidArgumentValueError(
+                '--set-localdns must be specified when using --localdns-config or --localdns-mode.'
+            )
+        if set_localdns:
+            if not (config or mode):
+                raise InvalidArgumentValueError(
+                    '--set-localdns requires either --localdns-config or --localdns-mode.'
+                )
+            if config and mode:
+                raise MutuallyExclusiveArgumentError(
+                    'Cannot specify both --localdns-config and --localdns-mode.'
+                )
+            if config:
+                if not isinstance(config, str) or not os.path.isfile(config):
+                    raise InvalidArgumentValueError(
+                        f"{config} is not a valid file, or not accessible."
+                    )
+                profile = get_file_json(config)
+                if not isinstance(profile, dict):
+                    raise InvalidArgumentValueError(
+                        f"Error reading local DNS config from {config}."
+                        "Please provide a valid JSON file."
+                    )
+                return profile
+            if mode:
+                if mode not in [
+                    CONST_LOCAL_DNS_MODE_REQUIRED,
+                    CONST_LOCAL_DNS_MODE_PREFERRED,
+                    CONST_LOCAL_DNS_MODE_DISABLED
+                ]:
+                    raise InvalidArgumentValueError(f"Invalid local DNS mode: {mode}")
+                # Return a minimal profile dict with just the mode
+                return {"mode": mode}
```

Comment:
Is there a model associated with this profile? If so, it's best to initialize the model type and configure its properties accordingly

Reply:
It should be inside the agentpool - I think I was able to match it up with the localDNSProfile. However, I had a quick question - do I need to also handle the parsing of the components in the json file?

I.e match up each of the values from the file with the data model?

Reply:
that depends, the purpose of conversion is to do a client-side validation to ensure that the input provided by the user meets the expectations of the model. Otherwise, you can also keep the dict form and hand it over to the server for validation.